src/main/scala/com/lambtors/poker_api/module/poker/application/turn/TurnCardAdder.scala

```diff
@@ -6,42 +6,41 @@ import com.lambtors.poker_api.module.poker.domain.error.{
   TurnNotPossibleWhenItIsAlreadyGiven
 }
 import com.lambtors.poker_api.module.poker.domain.{PlayerRepository, PokerGameRepository}
-import com.lambtors.poker_api.module.poker.domain.model.GameId
+import com.lambtors.poker_api.module.poker.domain.model.{Card, GameId, Player}
 import com.lambtors.poker_api.module.shared.domain.DeckProvider
-
 import scala.concurrent.{ExecutionContext, Future}
 
 final class TurnCardAdder(
-  repository: PokerGameRepository,
-  playerRepository: PlayerRepository,
-  deckProvider: DeckProvider
+    repository: PokerGameRepository,
+    playerRepository: PlayerRepository,
+    deckProvider: DeckProvider
 )(implicit ec: ExecutionContext) {
-  def add(gameId: GameId): Future[Unit] = repository.search(gameId).map(
-    gameOpt => if (gameOpt.isEmpty) {
-      throw PokerGameNotFound(gameId)
-    } else {
-      val game = gameOpt.get
-
-      if (game.tableCards.length > 3) {
-        throw TurnNotPossibleWhenItIsAlreadyGiven(gameId)
-      } else if (game.tableCards.length < 3) {
-        throw TurnNotPossibleWhenFlopIsNotGiven(gameId)
-      } else {
-        val cards = deckProvider.provide()
-
-        playerRepository.search(gameId).flatMap(
-          players => {
-            val playersCards = players.flatMap(player => List(player.firstCard, player.secondCard))
-            val playerCardsWithCardsAtTheTable = playersCards ++ game.tableCards
+  def add(gameId: GameId): Future[Unit] =
+    repository
+      .search(gameId)
+      .flatMap(_.fold(Future.failed[Unit](PokerGameNotFound(gameId))) { game =>
+        if (game.tableCards.length > 3) {
+          throw TurnNotPossibleWhenItIsAlreadyGiven(gameId)
```

Comment:
do we still throw exceptions at this level? or future.failed better in the functional branch?

Reply:
ooppppps! I forgot it!